stat_posix.go

```diff
@@ -0,0 +1,117 @@
+// +build !plan9
+
+package sftp
+
+import (
+	"os"
+	"syscall"
+)
+
+const EBADF = syscall.EBADF
+
+func wrapPathError(filepath string, err error) error {
+	switch err.(type) {
```

Comment:
Now that I look at this, this could probably better be:
```
if errno, ok := err.(syscall.Errno); ok {
  return &os.PathError{Path: filepath, Err: errno }
}
return err
```

With something similar for the plan9 with `ErrorString` instead of `Errno`.

A good compiler would make both equivalent, so this really should be about what is more readable, and/or if we expect this type-switch to be further expanded.

Your input is desired regarding this matter. It’s probably fine either way.

Reply:
I agree that an if-statement is better here. It was using a switch in the original code, so I just left it like that.

I've converted it to an if-statement.